### ISSUE TITLE
Does this code correspond to the paper?

### ISSUE THREAD

**Issue by QK51643729** [htcr/sam_road#16]
As I can see in Figure2 , the input to the NMS is the two graphs extracted from the model, but in the code dataset.py  I see that the NMS input comes from the ground truth. can you explain it?

**Comment by htcr** [comment]
Hi, the code does corresponds to the paper. As mentioned in the paper, during training, the model uses "teacher forcing", which adds noise to ground-truth to emulate the predicted vertices. You can find that in inference.py, the model first generates vertices from its predicted mask with NMS, then predicts topology based on those, no ground-truth involved during inference.

**Comment by QK51643729** [comment]
Ok, thank you for your timely reply. I don't know much about teacher force. Is it feasible to add noise to GT as training data?

**Comment by htcr** [comment]
It seems feasible according to our experiments. Rule of thumb is to emulate the predicted distribution well
